src/runner.rs

```diff
@@ -1,21 +1,41 @@
-use crate::constants::EXIT_FAILURE;
+use crate::clients::get_quarantine_bulk_test_status;
+use crate::constants::{EXIT_FAILURE, EXIT_SUCCESS};
 use crate::scanner::{BundleRepo, FileSet, FileSetCounter};
-use crate::types::{RunResult, Test};
+use crate::types::QuarantineBulkTestStatus;
+use crate::types::{QuarantineRunResult, RunResult, Test};
 use junit_parser;
 use std::fs::metadata;
 use std::process::Command;
 use std::process::Stdio;
 use std::time::SystemTime;
+use tokio_retry::strategy::ExponentialBackoff;
+use tokio_retry::Retry;
 
 pub async fn run_test_command(
     repo: &BundleRepo,
     command: &String,
     args: Vec<&String>,
-    output_paths: Vec<&String>,
+    output_paths: Vec<String>,
     team: Option<String>,
     codeowners_path: Option<String>,
 ) -> anyhow::Result<RunResult> {
     let start = SystemTime::now();
+    let exit_code = run_test_and_get_exit_code(command, args).await?;
+    log::info!("Command exit code: {}", exit_code);
+    let (file_sets, _file_counter) =
+        get_files(repo, output_paths, team.clone(), codeowners_path.clone())?;
+    let failures = if exit_code != EXIT_SUCCESS {
+        get_failures(file_sets, Some(start)).await?
+    } else {
+        Vec::new()
+    };
+    Ok(RunResult {
+        exit_code,
+        failures,
+    })
+}
+
+async fn run_test_and_get_exit_code(command: &String, args: Vec<&String>) -> anyhow::Result<i32> {
     let mut child = Command::new(command)
```

Comment:
I just noticed we use an unwrap on this. I think we should be using a try (`?` sigil) on this to bubble up the spawn error.